mavlink/qgroundcontrol
# UTM: Refactor Network Remote ID telemetry control based on vehicle state

Description
-----------
This PR refactors the UTMSPServiceController::networkRemoteID method to improve telemetry control logic:

* Removes position-based telemetry control logic
* Adds state-based control using MAV_STATE from HEARTBEAT messages
* Telemetry now starts when vehicle is in MAV_STATE_ACTIVE state
* Telemetry stops when vehicle transitions out of MAV_STATE_ACTIVE state
* Simplifies code flow and removes unnecessary position comparisons

Test Steps
-----------
1. Connect vehicle to QGC
2. Setup UTM Network Remote ID parameters (serialNumber, operatorID, flightID)
3. Verify telemetry starts when vehicle enters MAV_STATE_ACTIVE
4. Verify telemetry continues at 3Hz while in MAV_STATE_ACTIVE
5. Verify telemetry stops when vehicle exits MAV_STATE_ACTIVE state
6. Check logs to confirm proper state transitions and telemetry behavior

Checklist:
----------

Related Issue
-----------
N/A

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
